src/Payments.sol

```diff
@@ -0,0 +1,572 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./RateChangeQueue.sol";
+
+interface IArbiter {
+    struct ArbitrationResult {
+        bool approved;
+        uint256 modifiedAmount;
+        uint256 settledUntil;
+    }
+
+    function arbitratePayment(
+        uint256 railId,
+        uint256 proposedAmount,
+        uint256 fromEpoch,
+        uint256 toEpoch
+    ) external returns (ArbitrationResult memory result);
+}
+
+contract Payments is Initializable, UUPSUpgradeable, OwnableUpgradeable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using RateChangeQueue for RateChangeQueue.Queue;
+
+    struct Account {
+        address ownerAddress;
+        uint256 funds;
+        uint256 lockupCurrent;
+        uint256 lockupRate;
```

Comment:
Let's comment on the units here (and all the other fields). E.g. lockup rate is attotoken/epoch.

Reply:
@Stebalien Yeah plan is to raise a subsequent PR for comprehensive docs for both state and functions once we've locked in a meaningful draft.